spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/filters/AbstractRateLimitFilter.java

```diff
@@ -65,12 +65,15 @@ public boolean shouldFilter() {
             return false;
         }
 
-        if (originIsOnDenyRequest(request)) {
-            int responseStatusCode = properties.getDenyRequest().getResponseStatusCode();
-            ctx.setResponseStatusCode(responseStatusCode);
-            ctx.setSendZuulResponse(false);
+        if (shouldBypassRequestLocation(request)) {
+            return false;
+        }
+
+        if (shouldDenyRequestLocation(request)) {
+            ctx.setResponseStatusCode(HttpStatus.FORBIDDEN.value());
+            ctx.put(RATE_LIMIT_EXCEEDED, "true");
 
-            throw new RateLimitExceededException(HttpStatus.valueOf(responseStatusCode));
+            throw new RateLimitExceededException(HttpStatus.FORBIDDEN);
```

Comment:
It feels overly restrictive to have the status be fixed. It does make sense to have it return a 403 Forbidden but maybe some users want to return something different like 404 or 418. 
Not really a dealbreaker but just food for thought.

Reply:
I've added a property to make it configurable